zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -589,6 +600,8 @@ private void processEvent(Object event) {
                         ((AsyncCallback.EphemeralsCallback) lcb.cb).processResult(lcb.rc, lcb.ctx, null);
                     } else if (lcb.cb instanceof AsyncCallback.AllChildrenNumberCallback) {
                         ((AsyncCallback.AllChildrenNumberCallback) lcb.cb).processResult(lcb.rc, lcb.path, lcb.ctx, -1);
+                    } else if (lcb.cb instanceof AsyncCallback.MultiCallback) {
```

Comment:
It seems we don't call `queueCallback` anywhere so a `LocalCallback` is never created. If so, we should delete such dead code instead of adapting it.

Reply:
Hmmm I just see the next change, so we active an unused method. Let me take a closer look.

Reply:
at the first glance, it makes sense for me... AFAICT: if the client would like to execute an "empty" multiop request, we just return immediately without even sending the request to the server. 

@eolivelli Do you think we would need to implement the similar logic for the C-client? (I see we have `TestMulti.cc` with a few unit tests, but don't see any test for empty multi-request. Also I don't see in the c code that this specific case would be handled in the C client.) I think adding a new unit test for the C-client (and potentially make a fix) would make sense. Or at least register a follow-up jira for it.